org.eclipse.buildship.ui.test/src/main/groovy/org/eclipse/buildship/ui/launch/ClasspathSeparationTest.groovy

```diff
@@ -0,0 +1,206 @@
+package org.eclipse.buildship.ui.launch
+
+import spock.lang.IgnoreIf
+import spock.util.environment.OperatingSystem
+
+import com.gradleware.tooling.toolingclient.GradleDistribution
+
+import org.eclipse.core.runtime.NullProgressMonitor
+import org.eclipse.debug.core.DebugPlugin
+import org.eclipse.debug.core.ILaunch
+import org.eclipse.debug.core.ILaunchConfiguration
+import org.eclipse.debug.core.ILaunchConfigurationType
+import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy
+import org.eclipse.debug.core.ILaunchManager
+
+import org.eclipse.buildship.ui.test.fixtures.SwtBotSpecification
+
+// TODO (donat) enable the test when there's a Gradle snapshot release that includes the classpath separation changes
+// TODO (donat) adjust the Gradle distributions used by the test methods
+@IgnoreIf({ !OperatingSystem.current.isMacOs() })
+class ClasspathSeparationTest extends SwtBotSpecification {
+
+    def cleanup() {
+        DebugPlugin.default.launchManager.launchConfigurations.each { it.delete() }
+    }
+
+    def "Gradle doesn't supply scope information"() {
+        setup:
+        importAndWait(createSampleProject('sample-project'), GradleDistribution.forVersion('3.5'))
+
+        when:
+        launchAndWait(createJavaLaunchConfiguration('sample-project', 'pkg.Main'))
+
+        then:
+        assertConsoleOutputContains('pkg.Main available')
+        assertConsoleOutputContains('pkg.JunitTest available')
+        assertConsoleOutputContains('org.apache.commons.io.IOUtils available')
+        assertConsoleOutputContains('com.google.common.collect.ImmutableList available')
+        assertConsoleOutputContains('junit.framework.Test available')
+    }
+
+    def "Launch Java application from src/main/java"() {
+        setup:
+        importAndWait(createSampleProject('sample-project'), GradleDistribution.forLocalInstallation(new File('/Development/git/gradle/gradle/build/distributions/gradle-4.4-20171016160000+0000')))
+
+        when:
+        launchAndWait(createJavaLaunchConfiguration('sample-project', 'pkg.Main'))
+
+        then:
+        assertConsoleOutputContains('pkg.Main available')
+        assertConsoleOutputContains('pkg.JunitTest inaccessible')
+        assertConsoleOutputContains('org.apache.commons.io.IOUtils available')
```

Comment:
❌ This shouldn't be available, it's a `compileOnly` dependency. We should probably only use the `runtimeClasspath` to compute the `gradle-used-by-scope` attribute after all.

Reply:
I adjusted the implementation in Gradle core and updated the test assertions.